src/callbacks_step/time_series_dg2d.jl

```diff
@@ -68,6 +70,83 @@ function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh,
     return element_ids
 end
 
+function get_elements_by_coordinates!(element_ids, coordinates,
+                                      mesh::UnstructuredMesh2D,
+                                      dg, cache)
+    if length(element_ids) != size(coordinates, 2)
+        throw(DimensionMismatch("storage length for element ids does not match the number of coordinates"))
+    end
+
+    # Reset element ids - 0 indicates "not (yet) found"
+    element_ids .= 0
+    found_elements = 0
+
+    # Iterate over all elements
+    for element in eachelement(dg, cache)
+
+        # Iterate over coordinates
+        for index in 1:length(element_ids)
+            # Skip coordinates for which an element has already been found
+            if element_ids[index] > 0
+                continue
+            end
+
+            # Construct point
+            x = SVector(ntuple(i -> coordinates[i, index], ndims(mesh)))
```

Comment:
I am surprised that you can actually use an `SVector` for this :open_mouth:

Reply:
I didn't know this was possible either. I just copied this strategy from Felipe who implemented this callback on the `TreeMesh`.

Reply:
This `SVector` is constructed from an `NTuple` generated by `ntuple`. That's fine and the same strategy we use elsewhere, e.g.,

https://github.com/trixi-framework/Trixi.jl/blob/4bf61a0fd3abb21b457ea9d9ee2c19c835d018c3/src/solvers/dg.jl#L538-L550